Library/Homebrew/os/linux/elf.rb

```diff
@@ -126,7 +132,22 @@ def initialize(path)
 
     private
 
+    def default_on_exception(val, exception)
```

Comment:
cc @mistydemeo and @sjackman for opinions, but my personal preference is for not having a helper here and instead using Ruby's `expr rescue default` syntax.

For example:

```ruby
patcher.soname rescue nil
```

Reply:
that's new to me, thanks!

Reply:
I think we're stuck with this helper because of (reasonable) RuboCop rules; maybe it belongs in one of the `utils` files?

Reply:
if other cases for it pop up, sure we'll move into utils with a (better) name. okay?

Reply:
I'd personally prefer that it gets moved in this PR, to avoid future churn/simplifying future `git blames`.

Reply:
what do u suggest the name for the method ?

Reply:
@MikeMcQuaid any suggestions here?

Reply:
I think it's worth inlining two `begin` blocks. For something that's used twice I don't think it's worth splitting into a method which makes the existing code harder to follow.

Reply:
@MikeMcQuaid done, and tested. can u have one more go through whole thing?